onnxruntime/core/providers/cpu/math/element_wise_ops.cc

```diff
@@ -259,12 +259,12 @@ REG_ELEMENTWISE_VERSIONED_TYPED_KERNEL(Sqrt, 6, 12, double, Sqrt);
 REG_ELEMENTWISE_TYPED_KERNEL(Sqrt, 13, float, Sqrt);
 REG_ELEMENTWISE_TYPED_KERNEL(Sqrt, 13, double, Sqrt);
 
-const auto supported_pow7_types = BuildKernelDefConstraintsFromTypeList<Pow7Types>();
-const auto enabled_pow7_types = BuildKernelDefConstraintsFromTypeList<EnabledPow7Types>();
-const auto supported_pow12_base_types = BuildKernelDefConstraintsFromTypeList<Pow12BaseTypes>();
-const auto supported_pow12_exp_types = BuildKernelDefConstraintsFromTypeList<Pow12ExpTypes>();
-const auto enabled_pow12_base_types = BuildKernelDefConstraintsFromTypeList<EnabledPow12BaseTypes>();
-const auto enabled_pow12_exp_types = BuildKernelDefConstraintsFromTypeList<EnabledPow12ExpTypes>();
+#define supported_pow7_types (BuildKernelDefConstraintsFromTypeList<Pow7Types>())
```

Comment:
This change is to change a runtime initialization to compile time, to avoid initialization order problems.

Reply:
not that excited about adding more macros - these names don't look like macros either.
for these, can we just put the macro content where it is used? it might be nice to shorten the name BuildKernelDefConstraintsFromTypeList() if that's too unwieldy.

Reply:
Sure, I will make the change.
My view: even though macros look ugly. But when you don't have a better replacement, performance and correctness are still better than readiness. One bad example is eigen: whenever you include any eigen  header file, it will generate a lot global variables in that source file and run the initializers when your process starts.  onnxruntime has thousands of  such useless const variables in onnxruntime.dll. I admit the author of eigen knows C++ much much better than me, but if he had chosen using macros instead of "const xxx" in these cases,  it will save a lot!